client/src/lib.js

```diff
@@ -0,0 +1,135 @@
+import * as API from "./api.js"
+import CID from "multiformats/cid"
+
+
+
+/**
+ * @implements {API.Service}
+ */
+export default class NFTStore {
+  /**
+   * @param {Object} options
+   * @param {string} options.token
+   * @param {URL} [options.endpoint]
+   */
+  constructor({ token, endpoint = new URL("https://api.nft.storage") }) {
+    /** @readonly */
+    this.token = token
+    /** @readonly */
+    this.endpoint = endpoint
+  }
+
+  /**
+   * @private
+   * @param {string} token
+   */
+  static auth(token) {
+    return { Authorization: `Bearer ${token}` }
+  }
+  /**
+   * @param {API.Service} service
+   * @param {Blob} blob
+   */
+  static async storeBlob({ endpoint, token }, blob) {
+    const url = new URL("/", endpoint)
+    const request = await fetch(url.toString(), {
+      method: "PUT",
+      headers: NFTStore.auth(token),
+      body: blob
+    })
+    const { cid } = await request.json()
+
+    return CID.parse(cid)
+  }
+  /**
+   * @param {API.Service} service
+   * @param {Iterable<File>} files
+   */
+  static async storeDirectory({ endpoint, token }, files) {
+    const url = new URL("/", endpoint)
+    const body = new FormData()
+    for (const file of files) {
+      body.append("file", file, file.name)
+    }
+
+    const request = await fetch(url.toString(), {
+      method: "POST",
+      headers: NFTStore.auth(token),
+      body,
+    })
+    const { cid } = await request.json()
+
+    return CID.parse(cid)
+  }
+
+  /**
+   * @param {API.Service} service
+   * @param {CID} cid
+   * @returns {Promise<API.StatusResult>}
+   */
+  static async status({ endpoint, token }, cid) {
+    const url = new URL(`/${cid}`, endpoint)
+    const request = await fetch(url.toString(), {
```

Comment:
Can you pass the url object directly to fetch?

Reply:
Type checker won’t like it because type signature expects string not an URL instance.